sdk/lib/opentelemetry/sdk/trace/export/batch_sampled_span_processor.rb

```diff
@@ -21,7 +22,92 @@ module Export
         # worker thread that exports the spans to wake up and start a new
         # export cycle.
         class BatchSampledSpanProcessor
-          # TODO
+          SCHEDULE_DELAY_MILLIS = 5000
+          MAX_QUEUE_SIZE = 2048
+          MAX_EXPORT_BATCH_SIZE = 512
+
+          def initialize(exporter:, schedule_delay: SCHEDULE_DELAY_MILLIS, max_queue_size: MAX_QUEUE_SIZE, max_export_batch_size: MAX_EXPORT_BATCH_SIZE)
+            @exporter = exporter
+            @mutex = Mutex.new
+            @condition = ConditionVariable.new
+            @keep_running = true
+            @delay = schedule_delay
+            @max_queue_size = max_queue_size
+            @batch_size = max_export_batch_size
+            @spans = []
+            @thread = Thread.new { work }
+          end
+
+          # does nothing for this processor
+          def on_start(span)
+            # noop
+          end
+
+          # adds a span to the batcher, threadsafe may block on lock
+          def on_end(span)
```

Comment:
Whose responsibility is it to check if the span was sampled? Should that happen here, or further up?

Reply:
@fbogsany `BatchRecordingEventsSpanProcessor` ?

Reply:
I'm proposing just calling it the `BatchSpanProcessor`.